### ISSUE TITLE
No lora, No GPS fix

### ISSUE THREAD

**Issue by JordyGit** [wklenk/lmic-rpi-lora-gps-hat#6]
Hello,

For school project we have to make a sort of gps tracker with raspberry pi 3 model b and dragino lora/gps hat (v1.1). We have never used this before so everything is new for us.

Until now i tried to complete http://wiki.dragino.com/index.php?title=Getting_GPS_to_work_on_Raspberry_Pi_3_Model_B to set up the NMEA sentences from gps sensor. Until now i have only had empty NMEA sentences (with sudo cat /dev/ttyS0 in terminal) for hours with everything placed next to my window. (Maybe i use wrong antenna's, teachers gave this to use separately, otherwise there is nothing written on the antennas.) 
But between the empty ,,,,'s the time and date are perfectly correct. (does that mean something?)

Then i have tried for lora a few githubs, none of them did work without Failures. Except yours does, but after trying connecting for a few hours i still could not Join (didn't see JOINED, it's in a loop of trying to connect (EV_TXSTART,sending,retrieving,waiting,retrieving,waiting,EV_TXSTART,sending,...)) . 
Example "hello" does work fine. For example "join": yes , I have filled in /examples/join/main.c with our three TTN codes. Device is set as OTAA as it should.

For now we have soldered pin 22 on 24 because it's needed mentioned at http://wiki.dragino.com/index.php?title=Lora/GPS_HAT , for now we also have connected pin 16 to and 18 to because they did that to. (cut these connections doesnt change something to gps or lora connection, have tried that :) )

I have been trying to connect to lora from my house, which is at straight line at max 4km away from my school building, where they have a lorawan antenna at the top of the 7the floor.

I hope you could help us with this? You are the one who knows the best how this works because you made it. We would really appriciate it.

Kind Regards
Jordy from University KULeuven campus Brugge (Belgium)

**Comment by wklenk** [comment]
Sorry I never tried out the GPS functionality, but as you already read at the dragino.com site, it seems somehow complicated to get it working on version 3 of RPi.

> For example "join": yes , I have filled in /examples/join/main.c with our three TTN codes. Device is set as OTAA as it should.

It is kind of hard to do remote diagnosis, but I will try:
* The LMIC library randomly chooses one of the available transmission frequencies, and also tries to communicate using not only **one** spreading factor SF. For this reason, it is crucial that you have a fully fledged LoRaWAN gateway that supports the full LoRaWAN standard. If you just have a single-channel gateway with limited functionality, the communication probably won't work (without further modifications).  To go on with diagnosis, you should please tell we want brand and type of gateway you are using.
* Hopefully you took into regards that - when taking the "Application Router ID" and "Unique Device ID" you really have to use the *right* byte order, which is "least significant byte first" for these two configuration settings. I think the UI at TTN lets you choose different representations of these values when you copy and paste them from there. If you like, you can send my screenshots from both your code (the part where you define these values) and from TTN. My email adress is wolfgang.klenk at gmail.com.

**Comment by 3v3ryb0dy** [comment]
I have mailed you. Thank you for your fast respons.

**Comment by xiphias256** [comment]
@JordyGit Any updates to this issue? It seems to be related to #11

**Comment by JordyGit** [comment]
Nope, we gave up on the raspberry pi because we also had a deadline and the
raspberry also did crash a lot somehow (it was a new one).
It’s already a half year ago, i don’t really remember everything but we
used an arduino board with lorawan, and we could get that one working. I
send you some pictures and stuff attached to this email about our project.
(Pictures at the end part of the powerpoint you can see what we used, two
sensor boards (one with multiple sensors on it) and a wireless charging
module) It was a part of our task to tell everything about everything, like
how that little sensor works, or how lorawan works, we did this in dutch, i
attached it but i am pretty sure there could be mistakes in there because
we are not professionals, we were also new to all of this.
We were able to view our data on cayenne, our own php website, and a
gps-server website with all the sensordata (temperature, ...) to show
lorowan/TTN is  new good way to handle this data for what our needs where.
Our teachers were very happy about what we reached at the end, and it
resulted as a 18/20 for me as the project leader. :)

Kind regards
Jordy

 PWP V3 Officieel.pptx
<https://drive.google.com/file/d/1wXXSs5WbOAw5hzGEgTHMZ-CWPEsLM6-L/view?usp=drive_web>

 Projectwerk Intelligente Haven - Smart Logistic...
<https://drive.google.com/file/d/14KUpK-Vhrn-rpq2nFU1vRVi2gNNpVMj6/view?usp=drive_web>

 poster V3.1.pptx
<https://drive.google.com/file/d/18DqhnWVeTYGXTgbAO222gtj9CGmN0xrl/view?usp=drive_web>


Op do 23 aug. 2018 om 23:48 schreef xiphias256 <notifications@github.com>:
